plugins/inputs/kernel/kernel.go

```diff
@@ -28,27 +28,40 @@ var (
 )
 
 type Kernel struct {
+	ConfigCollect []string `toml:"collect"`
+
+	optCollect      map[string]bool
 	statFile        string
 	entropyStatFile string
+	ksmStatsDir     string
+}
+
+func (k *Kernel) Init() error {
+	k.optCollect = make(map[string]bool)
```

Comment:
We can preallocate the map here...
```suggestion
	k.optCollect = make(map[string]bool, len(k.ConfigCollect))
```

Reply:
Good one. It did cross my mind, but didn't really think about it.